checkpoint_1: testing_protocols

IgorZhiltsoff opened the pull request:
Implemented 4/4 testing protocols + 2 tests for each
Displayed a UML diagram with testing protocols hierarchy

IgorZhiltsoff committed:
Revert "Fixed"

This reverts commit d7374b61c16375ae908a9d661bea76b272c75bb2.

IgorZhiltsoff committed:
Implemented 4/4 testing protocols; Supplied UML class diagram

IgorZhiltsoff committed:
Got rid of warning

IgorZhiltsoff committed:
Brushed up UML

IgorZhiltsoff committed:
Added rendered diagram (png+svg)

IgorZhiltsoff committed:
Added rendered diagram (png+svg)

IgorZhiltsoff committed:
Merge remote-tracking branch 'origin/testing_protocols' into testing_protocols

IgorZhiltsoff committed:
Added diagram

IgorZhiltsoff commented:
TP: Last commit before 1st CHECKPOINT DEADLINE: [da81285](https://github.com/IgorZhiltsoff/TPy_project/pull/1/commits/da8128571f9948b1ffb3decb7c27add7362b5f74)

IgorZhiltsoff committed:
Added documentation for testing protocols; added new UML class diagram; drafted up new classes in testing_protocols.py; added task manager

IgorZhiltsoff committed:
Passed UserSubmittedData to check, instead of path_to_src + submission_id

IgorZhiltsoff committed:
check return type: str -> Verdict

IgorZhiltsoff committed:
Implemented ProblemData and ExtendedSubmission

IgorZhiltsoff committed:
moved convert_to_executable, conversion_opts, command_line_opts to TestingProtocols

IgorZhiltsoff committed:
Added yet another UML diagram; proposed __init__ for TestingProtocols

IgorZhiltsoff committed:
Added __init__(...) to TestingProtocols
Implemented ProgrammingLanguageData
Edited README.md
Enhanced puml diagram

IgorZhiltsoff committed:
Enforced ProgrammingLanguageData

IgorZhiltsoff committed:
Enforced pathlib
Started safeguarding files from collisions
Made path naming more consistent (got rid of *_location names)
Annotated ProgrammingLanguageData

IgorZhiltsoff committed:
Randomized decrease delta in collision processor; started implementing language support objects

IgorZhiltsoff committed:
tmp

IgorZhiltsoff commented:
PYTHON: Last commit before 1st REVIEW DEADLINE: [97e20c4](https://github.com/IgorZhiltsoff/TPy_project/pull/1/commits/97e20c49be73b8d33df1efd96881f2fa6ac79e24)
PYTHON: Next commit after DEADLINE with normal README.md: [230c3a2](https://github.com/IgorZhiltsoff/TPy_project/pull/1/commits/230c3a2104fcb3f4334865d50d2eba93de576913)

IgorZhiltsoff committed:
Implemented C++{11, 14, 17, 20} support

IgorZhiltsoff committed:
Implemented Haskell and Python support WITHOUT TESTS

UML update:
conversion ret type: Path → int - to trace CE

IgorZhiltsoff committed:
intermediate commit:
started testing python3
started refactoring conversion return type
edited readme

IgorZhiltsoff committed:
Refactored convert_to_executable return type: str -> int
Can handle CE

IgorZhiltsoff committed:
Got rid of excessive diagrams

IgorZhiltsoff committed:
Added rendered PlantUML diagrams

IgorZhiltsoff committed:
Added language_support.py contents to UML

IgorZhiltsoff committed:
Debugged language support for Python and tested it
Refactored protocols a bit

IgorZhiltsoff committed:
upd readme

IgorZhiltsoff committed:
Improved diagram

VTroyanGolovyan commented on TestingSystem/TestingProtocols/.idea/TestingProtocols.iml:
```diff
@@ -0,0 +1,10 @@
+  <component name="NewModuleRootManager">
+    <content url="file://$MODULE_DIR$">
+      <excludeFolder url="file://$MODULE_DIR$/venv" />
+    </content>
```
мусор

VTroyanGolovyan commented on TestingSystem/TestingProtocols/.idea/inspectionProfiles/profiles_settings.xml:
```diff
@@ -0,0 +1,6 @@
+<component name="InspectionProjectProfileManager">
+  <settings>
+    <option name="USE_PROJECT_PROFILE" value="false" />
```
мусор

VTroyanGolovyan commented on TestingSystem/TestingProtocols/.idea/misc.xml:
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="ProjectRootManager" version="2" project-jdk-name="Python 3.10 (classes)" project-jdk-type="Python SDK" />
```
мусор

VTroyanGolovyan commented on TestingSystem/TestingProtocols/sample_tests/test_python3.py:
```diff
@@ -0,0 +1,64 @@
+            programming_language_data=python_data.python3_data
+        )
+
+        self.assertEqual('AC', inout.check(user_submitted_data=correct).msg)
```
у тебя эта штука копипастой по всему проекту, подумай как от этого отойти

VTroyanGolovyan reviewed (approved):
Я не нашел где у тебя защита реализована.

нужно убрать мусор, и посмотреть внимательно на копипасту одинаковых кусков кода

пока ставлю 9

SokG2000 commented on TestingSystem/TestingProtocols/sample_tests/test_python3.py:
```diff
@@ -0,0 +1,64 @@
+import unittest
+from testing_protocols import InputOutput, InputCustomChecker, RandomInputCustomChecker, LimitedWorkSpace, \
```
Добейся того чтобы import работал при запуске файла с тестами интерпретатором питона или сделай инструкцию по запуску тестов. Разумеется при этом не надо ломать структуру проекта (например, перекладывать все файлы в одну папку). Наверное, самый правильный вариант - разобраться как работает import в python и как импортировать файлы из другой папки.

IgorZhiltsoff replied:
Добавил bash-скрипты, запускающие тесты (TestingSystem/TestingProtocols/run_*_tests.py)
(Команда PYTHONPATH=<src> python3 script.py запускает скрипт с кастомным путем для поиска файлов)

SokG2000 commented on TestingSystem/TestingProtocols/testing_protocols.py:
```diff
@@ -0,0 +1,301 @@
+        path_to_solution_output = TestingProtocol.generate_output_path(user_submitted_data.submission_id)
+
+        # ITERATE OVER TESTS
+        test = 0
```
Лучше for tests, path_to_input_file in enumerate(self.input_output_paths_dict)

SokG2000 reviewed (changes requested):
Напиши простую программу, использующую эту библиотеку.

IgorZhiltsoff committed:
Refactored .py's - turned into packages

IgorZhiltsoff committed:
Refactored names

IgorZhiltsoff committed:
Got rid of copy&paste in tests

IgorZhiltsoff committed:
Implemented handy i/f to prevent copy&paste

IgorZhiltsoff committed:
Enforced enumerate

IgorZhiltsoff committed:
Added scripts for running tests: TestingSystem/TestingProtocols/run_*_tests.sh

IgorZhiltsoff committed:
Created UseCase and Activity diagrams for uploading problems

IgorZhiltsoff committed:
Working on the class diagram

IgorZhiltsoff committed:
Designed upload

IgorZhiltsoff committed:
Drew up custodian

IgorZhiltsoff committed:
About to unify dumping problems and submissions

IgorZhiltsoff committed:
Tested basic custodian i/f

IgorZhiltsoff committed:
Implemented full custodian

IgorZhiltsoff committed:
Organizing

IgorZhiltsoff committed:
started switching to snake_case

IgorZhiltsoff committed:
managed UploadProblemAsTeacher structure

IgorZhiltsoff committed:
About to get rid of Custodian.extend

IgorZhiltsoff committed:
Added append to custodian

IgorZhiltsoff committed:
In the middle of smth.......

IgorZhiltsoff committed:
Implemented smth

IgorZhiltsoff committed:
Debugged upload cmd line i/f; Works now!

IgorZhiltsoff committed:
implemented running script

IgorZhiltsoff committed:
Replaced full paths with their suffixes "TPy_project/..."

IgorZhiltsoff committed:
Replaced full paths with their suffixes "TPy_project/..."

IgorZhiltsoff committed:
Implemented submission wizard!

IgorZhiltsoff committed:
Added Makefile

IgorZhiltsoff committed:
Debugged!

IgorZhiltsoff committed:
Cleaned up

IgorZhiltsoff committed:
Brushed up

IgorZhiltsoff committed:
drew up dir structure

IgorZhiltsoff committed:
drew up root

IgorZhiltsoff committed:
drew up display_readme

IgorZhiltsoff committed:
Added:
- .md support
- problem names
- Makefile
- scripts

Refreshed problem base

IgorZhiltsoff committed:
Learned how to "embed" pdfs

IgorZhiltsoff committed:
Enabled title prepend

IgorZhiltsoff committed:
Implemented and debugged display_problem_info!

IgorZhiltsoff committed:
Added return link

IgorZhiltsoff committed:
Brushed up

IgorZhiltsoff committed:
Displayed list!

IgorZhiltsoff committed:
Displayed list with refs!

IgorZhiltsoff committed:
Brushing up

IgorZhiltsoff committed:
Brushing up2

IgorZhiltsoff committed:
Brushing up2

IgorZhiltsoff committed:
Added 'back to main page' link to problem list

IgorZhiltsoff committed:
Switched to Jinja!

IgorZhiltsoff committed:
Switched to Jinja! (problem_info)

IgorZhiltsoff committed:
Got rid of unnecessary aux functions

IgorZhiltsoff committed:
Upper right corner!

IgorZhiltsoff committed:
Reduced copy&paste in templates

IgorZhiltsoff committed:
Reduced copy&paste

IgorZhiltsoff committed:
Implemented GOTO problem!

IgorZhiltsoff committed:
drew up submit.html

IgorZhiltsoff committed:
Enabled file upload!!

IgorZhiltsoff committed:
Fixed bug

IgorZhiltsoff committed:
Started implementing submit page

IgorZhiltsoff committed:
Started implementing submit page

IgorZhiltsoff committed:
Started implementing submit page

IgorZhiltsoff committed:
Improved naming in testing_protocols

IgorZhiltsoff committed:
Safely added labels

IgorZhiltsoff committed:
Got cpp tests to run
test rand failed: TestingProtocol.__init__() got an unexpected keyword argument 'language_data'
test lws failed: SKIP != AC

IgorZhiltsoff committed:
Fixed rand test

IgorZhiltsoff committed:
Passed CXX tests

IgorZhiltsoff committed:
Passed PYTHON tests

IgorZhiltsoff committed:
Banished opts from TestingProtocol class to ExecutionAndConversionData

IgorZhiltsoff committed:
Added script for running tests

IgorZhiltsoff committed:
Refactored diagrams_TestingProtocols.puml

IgorZhiltsoff committed:
Rendered

IgorZhiltsoff committed:
a

IgorZhiltsoff committed:
Revert "a"

This reverts commit 890b6e6021035de3d879f602977e9cb5116ecb97.

IgorZhiltsoff committed:
to amend

IgorZhiltsoff committed:
Enabled pre-commit hook

IgorZhiltsoff committed:
Verdict message made enum!

IgorZhiltsoff committed:
Started enforcing tempfile

IgorZhiltsoff committed:
Got rid of inout-cust copy&paste (check_with_chosen_language_data)

IgorZhiltsoff committed:
Used tempfile in randcust

IgorZhiltsoff committed:
Enforced tempfile in lws

IgorZhiltsoff committed:
Got rid of subprocess where possible in testing_protocols

IgorZhiltsoff committed:
Got rid of subprocess where possible in testing_protocols

IgorZhiltsoff committed:
Made filenames compatible with Windows

IgorZhiltsoff committed:
Prettified context manager use

IgorZhiltsoff committed:
drew up template json

IgorZhiltsoff committed:
Refactored wizard: now uploads execution_and_conversion_data_set

IgorZhiltsoff committed:
Refactored diagram

IgorZhiltsoff committed:
Attempted to refactor submission wizard's retriever module. About to test

IgorZhiltsoff committed:
Debugged submission wizard

IgorZhiltsoff committed:
Implemented process_submission for ServerIF!

IgorZhiltsoff committed:
Brushed up!

IgorZhiltsoff committed:
Enabled make once again!

IgorZhiltsoff committed:
Drew up file structure for html templates

IgorZhiltsoff committed:
Drawing up HTMLs

IgorZhiltsoff committed:
Kept drawing up HTMLs

IgorZhiltsoff committed:
Almost finished with HTMLs

IgorZhiltsoff committed:
Finished with HTMLs (at least for now...)

IgorZhiltsoff committed:
Drew up helper_upload_problem module

IgorZhiltsoff committed:
In the middle of smth...

IgorZhiltsoff committed:
In the middle of smth...2

IgorZhiltsoff committed:
Can input TL and ML floats via CMDLine!

IgorZhiltsoff committed:
Submission wizard now adds TL and ML to ExecutionAndConversionData

IgorZhiltsoff committed:
Enabled execution as nobody

IgorZhiltsoff committed:
Enforced execution as nobody in other parts of code

IgorZhiltsoff committed:
Brushed up

IgorZhiltsoff committed:
Getting ready to refactor run_code to measure resources

IgorZhiltsoff committed:
Placed resource monitoring programme
SRC: https://github.com/pshved/timeout

IgorZhiltsoff committed:
Added timeout as submodule

IgorZhiltsoff committed:
First timeout enforcement

IgorZhiltsoff committed:
Enforced timeout in rest of the places

IgorZhiltsoff committed:
Added files for testing

IgorZhiltsoff committed:
Intermediate commit
Refactoring Verdict
Refactoring run_code

IgorZhiltsoff committed:
Enforced resource consumption record!

IgorZhiltsoff committed:
Enforced resource consumption record!

IgorZhiltsoff committed:
Tested TL!

IgorZhiltsoff committed:
Removed bugs, improved naming

IgorZhiltsoff committed:
Added RE test for C++

IgorZhiltsoff committed:
Added ML and SKIP to tests!

IgorZhiltsoff committed:
Improved naming

IgorZhiltsoff committed:
Added TL test to C++

IgorZhiltsoff committed:
Added ABORT test for timelimit

IgorZhiltsoff committed:
Added missed ABORT test for checker, added ML ABORT test

IgorZhiltsoff committed:
Banished Abort tests to separate suite
Improved naming

IgorZhiltsoff committed:
Added Abort RE test
Brushed up file system

IgorZhiltsoff committed:
Added coverage (coverage is pretty good)

IgorZhiltsoff committed:
Cleaned up

IgorZhiltsoff committed:
Cleaned up 2

IgorZhiltsoff committed:
Fixed few bugs

IgorZhiltsoff committed:
Fixed few bugs

IgorZhiltsoff committed:
Rearranged files

IgorZhiltsoff committed:
Edited README.md
Started drawing up problem upload server interface

IgorZhiltsoff committed:
Added execution and conversion data to form!

IgorZhiltsoff committed:
Fully drew up htmls?

IgorZhiltsoff committed:
Implemented I/F!

IgorZhiltsoff committed:
Implemented I/F for real now

IgorZhiltsoff committed:
Drew up files

IgorZhiltsoff committed:
Drew up problem processing algorithm
Minor refactoring

IgorZhiltsoff committed:
Got smth running (but not quite)

IgorZhiltsoff committed:
Need to fix some bugs in cmd; server OK

IgorZhiltsoff committed:
Fixed bugs in CMD, successful inout upload!

IgorZhiltsoff committed:
Tested inout!

IgorZhiltsoff committed:
Added output

IgorZhiltsoff committed:
Fixed bug with permissions

IgorZhiltsoff committed:
Fixed some other bugs. Tested lws!

IgorZhiltsoff committed:
Almost done

IgorZhiltsoff committed:
Almost done! Fixing WA on incust (799)

IgorZhiltsoff committed:
Fixed incust (799), struggling with randcust (708)

IgorZhiltsoff committed:
Finished

IgorZhiltsoff committed:
Brushing up

IgorZhiltsoff committed:
Brushing up2

IgorZhiltsoff committed:
Oops! Totally forgot about client! Fixed

IgorZhiltsoff committed:
Merge pull request #3 from IgorZhiltsoff/server_interface

checkpoint_3

IgorZhiltsoff committed:
Merge pull request #2 from IgorZhiltsoff/interface

check_point2: Interface
